src/operation/list_indexes/mod.rs

```diff
@@ -0,0 +1,71 @@
+use crate::{
+    bson::{doc, Document},
+    cmap::{Command, StreamDescription},
+    cursor::CursorSpecification,
+    error::Result,
+    index::IndexModel,
+    operation::{append_options, Operation},
+    options::ListIndexOptions,
+    Namespace,
+};
+
+use super::{CursorBody, CursorResponse};
+
+#[cfg(test)]
+mod test;
+
+pub(crate) struct ListIndexes {
+    ns: Namespace,
+    options: Option<ListIndexOptions>,
+}
+
+impl ListIndexes {
+    pub(crate) fn new(ns: Namespace, options: Option<ListIndexOptions>) -> Self {
+        ListIndexes { ns, options }
+    }
+
+    #[cfg(test)]
+    pub(crate) fn empty() -> Self {
+        Self {
+            ns: Namespace {
+                db: String::new(),
+                coll: String::new(),
+            },
+            options: None,
+        }
+    }
+}
+
+impl Operation for ListIndexes {
```

Comment:
According to the [retryable reads spec](https://github.com/mongodb/specifications/blob/master/source/retryable-reads/retryable-reads.rst#supported-read-operations), this operation is read-retryable, so we should implement the `retryability` method here to return `Retryability::Read`. This also reminded me that we're currently skipping index management operations in the v2 and unified runners, so we'll need to implement those operations. We don't need to do that as part of this PR, though; I filed RUST-987 to do as add-on work.

Reply:
Sounds good, I also overrode `Operation::selection_criteria` since the [enumerate indexes spec](https://github.com/mongodb/specifications/blob/master/source/enumerate-indexes.rst#replica-sets) says

> Drivers MUST run `listIndexes` on the primary node when in a replica set topology, unless directly connected to a secondary node in Single topology.

Reply:
I think you accidentally added it to `CreateIndexes` -- that one isn't retryable